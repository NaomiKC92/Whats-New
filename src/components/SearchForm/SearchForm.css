* {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
}

main {
  width: 100%;
}

form {
  align-items: baseline;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  padding: 15px 15px;
}

input {
  background-color: #141527;
  border: none;
  font-size: 25px;
  height: 70px;
  width: 50%;
}

form > button {
  background-color: #B56AF0;
  border-radius: 50px;
  border: none;
  color: #FFF;
  font-size: 20px;
  height: 50px;
  width: 180px;
}