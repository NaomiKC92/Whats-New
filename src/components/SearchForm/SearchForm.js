import React, {Component} from 'react';
import './SearchForm.css';

class SearchForm extends Component {
  constructor() {
    super();
    this.state={
      input: ''
    }
  }

  handleOnChange = e => {
    this.setState({[e.target.name]: e.target.value});
  }

  resetInput = () => {
    this.setState({input: ''})
  }

  handleClick = e => {
    e.preventDefault();
    const inputText = this.state.input.toLowerCase();
    this.props.filterNews(inputText);
    this.resetInput();
  }

  render() {
    return (
      <form>
        <input type='text' 
        name='input'
        placeholder='Search for news articles here!'
        value={this.state.input}
        onChange={e => this.handleOnChange(e)}
        />
        <button onClick={e => this.handleClick(e)}>Search Now</button>
      </form>
    )
  }

}

export default SearchForm;