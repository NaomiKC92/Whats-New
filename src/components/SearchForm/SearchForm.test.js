import React from 'react';
import ReactDOM from 'react-dom';
import SearchForm from './SearchForm';
import { shallow } from 'enzyme';

describe('SearchForm', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<SearchForm />)
    expect(wrapper).toMatchSnapshot()
  })
})