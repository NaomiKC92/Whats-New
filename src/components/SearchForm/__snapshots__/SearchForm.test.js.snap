// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm should match the snapshot 1`] = `
<form>
  <input
    name="input"
    onChange={[Function]}
    placeholder="Enter search here"
    type="text"
    value=""
  />
  <button
    onClick={[Function]}
  >
    Search News
  </button>
</form>
`;
